### PR TITLE
support sdxl dynamic shape vanilla, lora, dreambooth finetune

### PR DESCRIPTION
**Pull request by wcrzlh**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_multi_aspect.yaml [hunk 1]
```diff
@@ -124,8 +124,8 @@ optim:
 
 
 data:
-    per_batch_size: 1
-    total_step: 100000
+    per_batch_size: 6
+    total_step: 4000
```
Comment (CaitinZhao) [inline examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_multi_aspect.yaml:128]: 训练step会不会太少了，训练结果ok吗

File: examples/stable_diffusion_xl/gm/helpers.py [hunk 2]
```diff
@@ -110,6 +110,9 @@ def set_default(args):
     if args.max_device_memory is not None:
         context.set_context(max_device_memory=args.max_device_memory)
         context.set_context(memory_optimize_level="O1", ascend_config={"atomic_clean_policy": 1})
+    if args.dynamic_shape:
+        context.set_context(mempool_block_size="59GB")
+        context.set_context(jit_config={"jit_level": "O1"}, enable_graph_kernel=True)
```
Comment (CaitinZhao) [inline examples/stable_diffusion_xl/gm/helpers.py:115]: 设了O1就不用设 enable_graph_kernel 了

Comment (CaitinZhao) [inline examples/stable_diffusion_xl/gm/helpers.py:115]:
```python
if args.mempool_block_size is not None:
        context.set_context(mempool_block_size=args.mempool_block_size)
```
动态shape默认"59GB"

File: examples/stable_diffusion_xl/gm/helpers.py [hunk 3]
```diff
@@ -301,11 +304,9 @@ def create_model(
 
 
 def convert_sdxl_to_fp16(sdxl):
-    vae = sdxl.first_stage_model
```
Comment (CaitinZhao) [inline examples/stable_diffusion_xl/gm/helpers.py:304]: 为什么去掉？

File: examples/stable_diffusion_xl/gm/modules/attention.py [hunk 4]
```diff
@@ -12,10 +13,9 @@
 
 import mindspore as ms
 from mindspore import nn, ops
-from mindspore.ops._tracefunc import trace
 
 try:
-    from mindspore.nn.layer.flash_attention import FlashAttention
+    from mindspore.ops.operations.nn_ops import FlashAttentionScore as FlashAttention
```
Comment (CaitinZhao) [inline examples/stable_diffusion_xl/gm/modules/attention.py:18]: 这个公共改，这个pr不要改这个

File: examples/stable_diffusion_xl/gm/modules/diffusionmodules/denoiser_scaling.py [hunk 5]
```diff
@@ -22,7 +22,7 @@ def construct(self, sigma):
         c_skip = ops.ones_like(sigma)
         c_out = -sigma
         c_in = 1 / (sigma**2 + 1.0) ** 0.5
-        c_noise = sigma.copy()
+        c_noise = sigma.copysign(1.0)
```
Comment (CaitinZhao) [inline examples/stable_diffusion_xl/gm/modules/diffusionmodules/denoiser_scaling.py:25]: 为什么改？

File: examples/stable_diffusion_xl/gm/helpers.py (hunk 3 above)
Comment (wcrzlh) [reply examples/stable_diffusion_xl/gm/helpers.py:304]: 参考diffusers, vae设置为fp32推理精度会好一些。
https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_sdxl.py#L713

File: examples/stable_diffusion_xl/gm/modules/diffusionmodules/denoiser_scaling.py (hunk 5 above)
Comment (wcrzlh) [reply examples/stable_diffusion_xl/gm/modules/diffusionmodules/denoiser_scaling.py:25]: 适配最新mindspore

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_multi_aspect.yaml (hunk 1 above)
Comment (wcrzlh) [reply examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_multi_aspect.yaml:128]: 当前所用数据集ok

**Pull request closed without merging**